### PR TITLE
Expose encrypted shares unauthenticated

### PR DESCRIPTION
**Pull request by achamayou**
Resolve https://github.com/microsoft/CCF/pull/5137#discussion_r1147841052

- [x] fix schema test
- [x] update cadl
- [x] update `submit_recovery_share.sh`

### CODE REVIEW

**Comment by eddyashton** [comment]
Linking to related issue: #5084

File: src/node/rpc/member_frontend.h [hunk 1]
```diff
@@ -901,6 +901,54 @@ namespace ccf
...
+          auto rec_share = GetRecoveryShare::Out{
+            crypto::b64_from_raw(encrypted_share.value())};
+          ctx.rpc_ctx->set_response_header(
+            http::headers::CONTENT_TYPE, http::headervalues::contenttype::JSON);
+          ctx.rpc_ctx->set_response_body(nlohmann::json(rec_share).dump());
+          ctx.rpc_ctx->set_response_status(HTTP_STATUS_OK);
+          return;
+        };
+      make_endpoint(
+        "/recovery_share/{member_id}",
```
Comment (jumaffre) [inline src/node/rpc/member_frontend.h:947]:
```suggestion
        "/encrypted_recovery_share/{member_id}",
```

Comment (achamayou) [reply src/node/rpc/member_frontend.h:947]: Good idea, gets rid of the question about symmetry too.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 [4.0.0-dev7]: https://github.com/microsoft/CCF/releases/tag/ccf-4.0.0-dev7
 
+### Deprecated
+
+- `GET /gov/recovery_share` is deprecated in favour of the unauthenticated `GET /gov/encrypted_recovery_share/{member_id}`.
+- The `submit_recovery_share.sh` script now takes a `--member-id` argument.
```
Comment (eddyashton) [inline CHANGELOG.md:15]:
```suggestion
- The `submit_recovery_share.sh` script now takes a `--cert` argument.
```

Currently takes a `--cert` argument and computes the `member-id`.

File: python/utils/submit_recovery_share.sh [hunk 3]
```diff
@@ -41,13 +41,34 @@ while [ "$1" != "" ]; do
...
+
+for item in "$@" ; do
+    if [ "$next_is_cert" == true ]; then
+        cert=$item
+        next_is_cert=false
+    fi
+    if [ "$item" == "--cert" ]; then
+        next_is_cert=true
+    fi
+done
```
[the remark this answers is not in the document]
Comment (achamayou) [reply python/utils/submit_recovery_share.sh:55]: Seems harmless to pass it on to curl?

[the remark this answers is not in the document]
Comment (achamayou) [reply python/utils/submit_recovery_share.sh:55]: It's a pain, that pass assumes the arguments are intercepted first, and everything else is handed over to curl. But --cert can come at any time. It's already broken if --help is passed later on.

I think this is fine for an illustrative sample, if we want a proper CLI we should find a library, expose a complete help etc.

[the remark this answers is not in the document]
Comment (eddyashton) [reply python/utils/submit_recovery_share.sh:55]: Huh, I thought we had a non-destructive parser pattern but it looks like we don't. So yeah, parse like this and it gets passed on to curl.

File: python/utils/submit_recovery_share.sh [hunk 4]
```diff
@@ -41,13 +41,34 @@ while [ "$1" != "" ]; do
...
+
 if [ -z "${member_enc_privk}" ]; then
     echo "Error: No member encryption private key in arguments (--member-enc-privk)"
     exit 1
 fi
 
+if [ -z "${cert}" ]; then
+    echo "Error: No user certificate in arguments (--cert)"
```
Comment (eddyashton) [inline python/utils/submit_recovery_share.sh:63]: We should add `--cert` to the `usage()` printed line.

**Review by eddyashton — APPROVED** [review]

**Pull request merged**